src/Native/System.Private.CoreLib.Native/pal_time.cpp

```diff
@@ -33,3 +34,223 @@ mach_timebase_info_data_t *InitializeTimebaseInfo()
     return g_isMachTimebaseInfoInitialized ? &g_machTimebaseInfo : nullptr;
 }
 #endif
+
+#if HAVE_CLOCK_MONOTONIC_COARSE || HAVE_CLOCK_MONOTONIC
+clockid_t GetMonotonicClockType(bool preferCoarseIfAvailable)
+{
```

Comment:
This could be simplified to:
```
#if HAVE_CLOCK_MONOTONIC_COARSE
    if (preferCoarseIfAvailable)
    {
        return CLOCK_MONOTONIC_COARSE;
    }
#endif
    return CLOCK_MONOTONIC;
```